proposals/public-suffix.md

```diff
@@ -0,0 +1,555 @@
+by default to be the last domain label of the domain name, or alternatively
+the domain name could be considered invalid.
+
+**Note:** it may be more performant to allow unknown suffixes and assume a single-label
+suffix by default, because it allows the following optimisation of the lookup algorithm:
+all single-label suffixes in the PSL can be excluded from the lookup, since they do not
+need to be matched specifically.
+
+#### 4. IDN
+
+The PSL uses Unicode, and contains International Domain Name (IDN) suffixes. Depending
+on the use case, when returning the *registrable domain / eTLD+1* for a domain name,
+either Unicode or Punycode may be preferred.
+
+**Note:** the [PSL algorithm](https://github.com/publicsuffix/list/wiki/Format#formal-algorithm)
+requires Punycode for the matching logic. Therefore a requirement to convert Punycode
+back to Unicode involves extra work. It may be preferable to avoid this in
+performance-sensitive use cases.
+
+#### Use Cases
+
+#### 1. Filter Requests by Organization
+
+This API is relevant to any extension that does both of the following:
+
+* acts on web requests automatically while the user browses
+* allows the user to choose which domain names the extension should act on,
+and which should be ignored by the extension
+
+In effect, such extensions allow users to create request filtering rules that
+restrict the extension to acting on only web requests having specific domain names.
+
+When choosing domain names for the request filter, users may want to be able to choose
+all domain names for an organization without having to enter them each in individually.
+The user may want to specify request filtering rules as follows:
+
+| # | Filtering rule type | User value | Examples of domains affected |
+|:-:|---------------------|:----------:|-----------------------------:|
+| 1 | organizational base domain and all possible subdomains | myorg.co.uk | myorg.co.uk<br>subdomain1.myorg.co.uk<br>subdomain2.myorg.co.uk |
+| 2 | organization name and all possible suffixes | myorg | myorg.co.uk<br>myorg.com<br>myorg.net |
+
+To support this use case, such extensions may:
+
+* automatically calculate the set of organization names and *registrable domain / eTLD+1*s
+from a user-specified set of domain names, and propose these as filtering rules for
+the user to choose
+* prevent users from mistakenly specifying a *public suffix / eTLD* as a filtering rule,
+believing it to be a *registrable domain / eTLD+1*, e.g. `co.uk` instead of `myorg.co.uk`
+* apply the users' filtering rules automatically while the user browses by determining
+the *registrable domain / eTLD+1* of each web request, and testing this using appropriate
+regexes to decide whether or not to act on the request
+
+##### PSL Requirements
+
+| PSL Feature            | Requirement | Discussion |
+|------------------------|-------------|------------|
+| Allow Private Suffixes | Yes & No    | some filters may require private suffixes, others ICANN-only |
```

Comment:
The PSL exists because filtering by organization requires the information on the PSL so only using the ICANN section seems just wrong here.

Reply:
Another reason why I dislike this is, that the distinction between an ICANN/IANA TLD and an eTLD is - in practice - poorly defined.
The "official" ICANN/IANA TLD is only the last label. There is an [IANA list](https://www.iana.org/domains/root/db) and another [list for gTLDs](https://www.icann.org/resources/registries/gtlds/v2/gtlds.json).

However, the PSL is trying to gather information from the official registrars and include that. To maintain the PSL we try to get in touch with the registrars to get them to update their "official" subdomains or to ask questions but we often fail.

Look at  the`.au` section of the list, they cannot agree what their TLDs are but they are in the ICANN section. `.bd` even has a wildcard `*.bd` entry in the ICANN section. `.co.uk` is in the ICANN section but the actual IANA TLD is still only `.uk`.